operational_analysis/toolkits/reanalysis_downloading.py

```diff
@@ -0,0 +1,369 @@
+for providing valuable input on portions of this code.
+
+To use this module to download data through the PlanetOS API, users must first create a PlanetOS
+account (https://data.planetos.com/plans). Once an account is created, an API key will be provided,
+which can be found in Account Settings. They API key should be saved in a text file, the location
+of which is passed as an argument to the downloading functions in this module. By default, it is
+assumed that the API key is saved in a file called "APIKEY" located in the OpenOA toolkits
+directory.
+
+More information about PlanetOS data products can be found here:
+http://docs.planetos.com/#planet-os-product-guide
+
+Alternatively, users can download MERRA-2 data directly from the NASA Goddard Earth Sciences Data
+and Information Services Center (GES DISC) and ERA5 data directly from the Copernicus Climate Data
+Store (CDS) service. However, this module does not currently contain functions for automating the
+downloading process.
+
+-Hourly MERRA-2 data can be downloaded directly from NASA GES DISC by selecting the
+"Subset / Get Data" link on the following webpage:
+https://disc.gsfc.nasa.gov/datasets/M2T1NXSLV_5.12.4/summary. Specific dates, variables, and
+coordinates can be selected using the OPeNDAP or GES DISC Subsetter download methods.
+
+-Hourly ERA5 data can be downloaded using either the CDS web interface or the CDS API, as explained
+here: https://confluence.ecmwf.int/display/CKB/How+to+download+ERA5. Data for specific dates,
+variables, and coordinates can be downloaded using the CDS web interface via the "Download data"
+tab here: https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels?tab=overview.
+Instructions for using the CDS Toolbox API to download ERA5 data programatically can be found here:
+https://cds.climate.copernicus.eu/toolbox/doc/how-to/1_how_to_retrieve_data/1_how_to_retrieve_data.html
+(note that the "reanalysis-era5-single-levels" dataset should be used).
+
+For reanalysis data downloaded directly from NASA or Copernicus, to import the data into an OpenOA
+PlantData object, used by the OpenOA analysis methods, the data should be re-saved as a single csv
+file with a separate row for each timestamp and columns corresponding to the variables of interest
+(including a timestamp column).
+
+Although any combination of available reanalysis variables can be downloaded, several of the OpenOA
+analysis methods require the eastward and northward wind components, temperature, and surface
+pressure (temperature and surface pressure can be used to calculate air density) over a 20-year period.
+"""
+
+import os
+import datetime
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import requests
+
+import operational_analysis.toolkits.met_data_processing as met
+
+
+# Maps OpenOA-standard reanalysis product names to PlanetOS datasets
+dataset_names = {"merra2": "nasa_merra2_global", "era5": "ecmwf_era5_v2"}
```

Comment:
I might have missed this before, but constants are typically in all caps in python. That said, we don't use constants much throughout OpenOA, so I could go either way on that convention. The main advantage would be being able to identify variables (constants) that sit outside the function more clearly, and signaling the intent of the variable (constant).

Reply:
I decided to change this dictionary constant to a function to add some string normalization and error handling.